apis/status/v1beta1/mutatorpodstatus_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"strings"
+
+	mtypes "github.com/open-policy-agent/gatekeeper/pkg/mutation/types"
+	"github.com/open-policy-agent/gatekeeper/pkg/operations"
+	"github.com/open-policy-agent/gatekeeper/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	MutationsGroup = "mutations.gatekeeper.sh"
+)
+
+// MutatorPodStatusStatus defines the observed state of MutatorPodStatus
+type MutatorPodStatusStatus struct {
+	// Important: Run "make" to regenerate code after modifying this file
+
+	ID string `json:"id,omitempty"`
+	// Storing the mutator UID allows us to detect drift, such as
+	// when a mutator has been recreated after its CRD was deleted
+	// out from under it, interrupting the watch
+	MutatorUID         types.UID `json:"mutatorUID,omitempty"`
+	Operations         []string  `json:"operations,omitempty"`
+	Enforced           bool      `json:"enforced,omitempty"`
```

Comment:
Is this a per pod value?
As I understand, this means that the Assign/AssignMetadata is enforced in the system (the mutation is valid and successfully added to the mutation system). Maybe this should be a property directly in the mutation status?

Reply:
Unfortunately a single pod can't really make that determination, we only know with 100% certainty when a mutator is enforced when all pods are reporting as enforced.

See this doc for my breakdown of the dynamics:

https://docs.google.com/document/d/1pyM3h5c8gh5W2eKP37qpge9jtEtRAs6Uv_sTyJttm3c/edit

Reply:
My bad, you're right on this one